test/test_core.py

```diff
@@ -2721,6 +2724,9 @@ def test_pthread_tls(self):
 
   @no_modularize_instance('uses global Module objecgt')
   def test_pthread_run_script(self):
+    if '-O0' in self.cflags and ('-flto' in self.cflags or '-flto=thin' in self.cflags):
```

Comment:
How about `if not self.is_optimizing()` (to catch the case where no `-O` flag is passed)